app/src/org/commcare/adapters/EntityListAdapter.java

```diff
@@ -429,7 +430,10 @@ public void loadCalloutDataFromSession() {
 
     public void saveCalloutDataToSession() {
         if (isFilteringByCalloutResult) {
-            CommCareApplication.instance().getCurrentSession().addExtraToCurrentFrameStep(SessionInstanceBuilder.KEY_ENTITY_LIST_EXTRA_DATA, calloutResponseData);
+            SessionWrapper session = CommCareApplication.instance().getCurrentSession();
+            session.removeExtraFromCurrentFrameStep(SessionInstanceBuilder.KEY_ENTITY_LIST_EXTRA_DATA);
+            session.addExtraToCurrentFrameStep(SessionInstanceBuilder.KEY_ENTITY_LIST_EXTRA_DATA,
```

Comment:
This means that the caller needs to know that they need to remove the extra first in order to add another. Why not leave the removal to the [addExtraTopStep](https://github.com/dimagi/commcare-core/blob/7cc09320ce920bf89b7b43255fd8850fb6c172c7/src/main/java/org/commcare/session/SessionFrame.java#L232) and [addExtra](https://github.com/dimagi/commcare-core/blob/7cc09320ce920bf89b7b43255fd8850fb6c172c7/src/main/java/org/commcare/suite/model/StackFrameStep.java#L162).

Reply:
true, got it.